### DISCUSSION TITLE
I want to build a little taller

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#17038] [Feature requests]
~~I want to be a baller~~

I want to use OpenRCT2 to make big mountain ranges. The existing height limit is not satisfying. I want to build taller.

RCT1 lovers want to build a little shorter. RCT1, through the use of cheats, could place objects at half height increments.

Both of these desires could be satisfied by widening the TileElement height field to 2 bytes (or even 1.5 nibbles would be satisfactory).

To prevent the need for a second new bit, clearance height would be stored as an offset and the getter/setter would convert. TODO: determine if any tile element gets a clearance height taller than 127

The tile has already been widened once, to 16 bytes. Some tile elements have don't have any bits to spare:
- path
- track

For path, maybe flags2 can union with slope direction? I don't know the values that they all use.

For track, I've thought about it a lot, and there is no simple answer on packing in more data.

**Comment by duncanspumpkin** [comment]
We spent a long time getting rid of the unions we are not going to bring in a new one.

**Comment by J-PIE-314** [comment]
The features you have suggested are now implemented. The height limits for land and water have been increased, and you can use the cheat "allow building track at invalid heights" to build paths and more in half height increments.

**Comment by spacek531** [comment]
This came up in discussion yesterday. The idea was brought up to widen base height to just 10 bits. With 10 bits, the height system can accommodate RCT1 heights and double the current height limit, which is plenty for any park that isn't trying to recreate a mountain. I know a few people who want to build a mountain but aren't, because the height limit is just too limiting, and this change might allow them that ability.

The current system of storing the clearance height would have to be modified to an offset from the base height, but this could be converted to and from absolute height using setters and getters. Research will have to be conducted to see if any parks would be affected by this change: are there any parks that have a clearance height that is 8 full height walls tall and higher? And, does lowering the clearances of existing tile elements change how the tile element or others are drawn?

As a point of reference, this (plus half a height unit) would be the new max height of TileElements after moving to a 10+6 bit base+offset system:

![image](https://github.com/user-attachments/assets/8e1cbf56-2ec1-4c15-96e6-c46cf5c1bf65)

**Reply by 733737** [reply]
To me it looks like a good size for custom scenery to be limited to. I've also seen large custom scenery broken into multiple pieces. I don't think large custom scenery is that popular, and it usually is limited to the size of the pyramid I believe.

**Reply by spacek531** [reply]
According to @danielmeinert this height limit is already enforced by the game, since scenery objects record the height in pixel units instead of height units--importantly, in a uint8_t. 8 pixel units make one height unit, and with a limit of 255, the tallest a scenery piece can be is 31 height units. Since it rounds down to the nearest height value, the maximum legal value is 248, but after the change it will be 252. This discrepancy is small, or it can be fixed by clamping the max height to 248.
